proofs/refiner.ml

```diff
@@ -204,14 +204,10 @@ let tclSHOWHYPS (tac : tactic) (goal: Goal.goal Evd.sigma)
 
 
 let catch_failerror (e, info) =
-  if catchable_exception e then Control.check_for_interrupt ()
-  else match e with
-  | FailError (0,_) ->
-      Control.check_for_interrupt ()
-  | FailError (lvl,s) ->
+  match e with
+  | FailError (lvl,s) when lvl > 0 ->
     iraise (FailError (lvl - 1, s), info)
-  | e -> iraise (e, info)
-  (** FIXME: do we need to add a [Errors.push] here? *)
+  | e -> Control.check_for_interrupt ()
```

Comment:
You changed the semantics here, in case of anomaly you still need to reraise it.

Reply:
I started from the assumption that one should not send anomalies to `tclZERO`. Therefore, `catch_failerror`, which is fed with a monad-level exception (using `tclOR` and cousins, in `Tacticals`), is not supposed to have to deal with anomalies. Isn't this invariant correct?

Reply:
It seems the best path would be to remove this function altogether, tho it doesn't need to be in this PR.

Reply:
> It seems the best path would be to remove this function altogether

Which function? `catch_failerror` or the call to `check_for_interrupt` inside `catch_failerror`?

For the former, we need it to decrease the `fail` level, so it needs to be kept afaicj. For the latter, we may try to remove it. It is called otherwise in `tclREPEAT`. The point is that I don't feel competent to evaluate whether this is enough often (for instance, it should then probably be called in `tclDO` and `tclFIRST` too).

Reply:
I mean `catch_failerror` altogether, but indeed that's not trivial to do.